src/ossia/protocols/midi/midi_protocol.hpp

```diff
@@ -33,27 +33,39 @@ struct OSSIA_EXPORT midi_info
   };
 
   midi_info() = default;
-  midi_info(Type t, std::string d, int p)
-      : type{t}, device{std::move(d)}, port{p}
+  midi_info(Type t, std::string d, std::string n, int p, bool v)
+      : type {t}
+      , device {std::move(d)}
+      , name {std::move(n)}
+      , port {p}
+      , isVirtual {v}
   {
   }
 
-  Type type{};
-  std::string device{};
-  int port{};
+  Type type {};
+  std::string device {};
+  std::string name {};
+  int port {};
+  bool isVirtual {};
 };
 
 class OSSIA_EXPORT midi_protocol final : public ossia::net::protocol_base
 {
 public:
-  explicit midi_protocol(ossia::net::network_context_ptr, libremidi::API api = libremidi::API::UNSPECIFIED);
-  explicit midi_protocol(ossia::net::network_context_ptr, midi_info, libremidi::API api = libremidi::API::UNSPECIFIED);
+  explicit midi_protocol(
+      ossia::net::network_context_ptr,
+      libremidi::API api = libremidi::API::UNSPECIFIED,
+      std::string peripheral_name = "ossia-midi");
```

Comment:
there is one small behaviour change here: before, it would create an ossia-in and an ossia-out thing depending on the direction, but now it creates ossia-midi no matter the direction.

I think that it'd actually be better to change the code so that it is not a default parameter anymore here, to force the user to choose a name (they can keep putting ossia-in / ossia-out if they need it because their setup expect it, for instance it's a name that they could have saved in a JACK patchbay or something like this

Reply:
removed default parameter value and refactored the constructors